### ISSUE TITLE
Possible problem with overriding of File.read in your lib/utils/file.rb?

### ISSUE THREAD

**Issue by vestedpr-dev** [starkbank/ecdsa-ruby#4]
As our tests were running in CI today we suddenly got a load of failures and quite obscure ones too.

starkbank-ecdsa was loaded into our app via the sendgrid-ruby library which, just released their version 6.3.0 today which, includes your gem.

My particular exception was thrown by a library, FFaker but the root problem came down to starkbank-ecdsa overriding Ruby's elemental File.read method in a way which I suspect is not fully compatible with all the ways this can be used:

```
undefined method `upcase' for {:mode=>"r:UTF-8"}:Hash
starkbank-ecdsa (0.0.2) lib/utils/file.rb:4:in `read'
ffaker (2.15.0) lib/ffaker/utils/module_utils.rb:21:in `const_missing'
ffaker (2.15.0) lib/ffaker/lorem.rb:19:in `words'
ffaker (2.15.0) lib/ffaker/lorem.rb:23:in `sentence'
ffaker (2.15.0) lib/ffaker/lorem.rb:32:in `block in sentences'
activesupport (5.2.4.3) lib/active_support/core_ext/range/each.rb:9:in `each'
activesupport (5.2.4.3) lib/active_support/core_ext/range/each.rb:9:in `each'
ffaker (2.15.0) lib/ffaker/lorem.rb:32:in `map'
ffaker (2.15.0) lib/ffaker/lorem.rb:32:in `sentences'
ffaker (2.15.0) lib/ffaker/lorem.rb:38:in `paragraph'
ffaker (2.15.0) lib/ffaker/lorem.rb:42:in `block in paragraphs'
activesupport (5.2.4.3) lib/active_support/core_ext/range/each.rb:9:in `each'
activesupport (5.2.4.3) lib/active_support/core_ext/range/each.rb:9:in `each'
ffaker (2.15.0) lib/ffaker/lorem.rb:42:in `map'
ffaker (2.15.0) lib/ffaker/lorem.rb:42:in `paragraphs'
```

The way this happens in the above stacktrace is this:

```
module FFaker
  module ModuleUtils
    include RandomUtils

    def k(arg)
      FFaker::ArrayUtils.const_array(arg)
    end

    def const_missing(const_name)
      if const_name =~ /[a-z]/ # Not a constant, probably a class/module name.
        super const_name
      else
        mod_name = ancestors.first.to_s.split('::').last
        data_path = "#{FFaker::BASE_LIB_PATH}/ffaker/data/#{underscore(mod_name)}/#{underscore(const_name.to_s)}"
        data = k File.read(data_path, mode: 'r:UTF-8').split("\n") # HERE THIS LINE
        const_set const_name, data
        data
      end
    end
```

This snippet: `File.read(data_path, mode: 'r:UTF-8')` -> `mode: 'r:UTF-8'`

In starkbank-ecdsa, `lib/utils/file.rb`:

```
class File

    def self.read(path, encoding="ASCII")
        file = File.open(path, :encoding => encoding.upcase)
        content = file.read
        file.close
        return content
    end

end
```

I'll be honest I get a little hazy but here you are overriding File.read which, is a really fundamental sort of thing ... and normally accepts more parameters in a variety of ways.

Perhaps check if you should add more namespacing there?

Thanks for your library!  Something to consider.

**Comment by childish-sambino** [comment]
Bump on this one. Approaching a delivery date and need to know if this will be looked at soon.

**Comment by cdottori-stark** [comment]
Just published the fix on version 0.0.3.

Sorry about the huge delay, our development team completely overloaded on these past few weeks and we ended up leaving our public libraries a bit unattended, but we'll focus on clearing all pending issues today e then resume our usually quicker maintenance.

Thanks for the feedback!
